### PR TITLE
Add more detail to PR recommendations

### PR DESCRIPTION
**Pull request by akash1810**
> **Note**
> This change can be [previewed here](https://github.com/guardian/recommendations/blob/325929187adfa71f92b83cf0fbadebcaecf35cbc/pull-requests.md).

## What is being ~recommended~ changed?
This change adds some information from [a Google Doc](https://docs.google.com/document/d/1bjmO0IbAUGhTCg33ySk3yaCcXkmnI1IQJXHJ6OyxQo0/edit), so that it's tracked in VCS.

❓ This change makes the recommendation significantly longer, and thus less succinct. Does this reduce its value?

## What's the context?
To avoid multiple sources of truth, this change is aimed at deleting a Google Doc.

### CODE REVIEW

**Review by shtukas — APPROVED** [review]

**Review by shtukas — APPROVED** [review]

**Review by bryophyta — APPROVED** [review]
There's some really great stuff here, thanks for adding it!

I see where you're coming from about longer guidance sometimes being less useful, either because it's harder to find the key information or because it can feel a bit overwhelming. A couple of options here might be 

1. to add a table of contents at the top of the page.
2. refactor some parts (e.g. discussion of the different types of review) to focus more on the recommendations, and link out to docs for the mechanics.

On (2), it feels like the main 'value add' of this document is to promote a positive and useful culture of reviewing. Given this, we could remove some of the factual information which isn't Guardian specific. (e.g. `Note: If a different reviewer has requested a code change using the “request change” option, you will not be able to approve the PR. It is effectively blocked until the requester has approved the PR.`) Perhaps linking to the Github docs would be a good substitute?

I don't think any of the above should be a blocker to this PR being merged though. I'd be happy to open separate PRs for (1) and (2) and see what people think.

File: pull-requests.md
Comment (mxdvl) [inline pull-requests.md:47]: Should we link to a good article on rewriting git history?

There's a good one from Github I cannot find and one @sndrs shared before I cannot remember the title of…

File: pull-requests.md
Comment (mxdvl) [inline pull-requests.md:130]: Rogue comma, mix of ampersand and not, use of capitalised Production, but PROD is used elsewhere.


```suggestion
A PR is not finished when you merge it - check your release in PROD after it's deployed! Although unit and integration tests can add a lot of confidence that your change is good, looking at your change in PROD is the ultimate and most important place to check your PR is delivering what's wanted.
```

File: pull-requests.md
Comment (mxdvl) [inline pull-requests.md:148]: I think this is a repetition. Do we want to lean into Markdown and implement footnotes?

**Review by kenoir — APPROVED** [review]

**Review by mxdvl — APPROVED** [review]
I think more organised guidance on good pull requests is better.

File: pull-requests.md
Comment (akash1810) [reply pull-requests.md:26]: I refer to these as Stacked PRs. There are a few good reads about this (for example [this](https://benjamincongdon.me/blog/2022/07/17/In-Praise-of-Stacked-PRs/) or [this](https://blog.logrocket.com/using-stacked-pull-requests-in-github/)).

Although I often use them, I do sometimes find them tricky. For example, if the history of a parent PR is rewritten, the child PR needs to be rebased so that it merges cleanly, however the rebase isn't always trivial.

For this reason, I'd be tempted to avoid _recommending_ them here, but I do agree, it'll be nice to make a note of this practice somewhere.

File: pull-requests.md
Comment (akash1810) [reply pull-requests.md:48]: Will merge this PR as is, but lets add the link once we find it.

File: pull-requests.md
Comment (sndrs) [reply pull-requests.md:26]: agree they're useful judiciously, but i find they often create noise for reviewers, even though they make a lot of sense the to PR raiser (which one do i need to review? which of these other PRs that are open and look like the need reviewing don't need reviewing yet?)

in a small team where everyone knows what's going on that's not a huge overhead. but in larger codebases, esp with lots of external contributors, it can be cumbersome

File: pull-requests.md
Comment (sndrs) [reply pull-requests.md:48]: was it this? https://tekin.co.uk/2019/02/a-talk-about-revision-histories

it was @akash1810 that pointed me to it :)

Comment (mxdvl) [reply pull-requests.md:48]: “It’s the circle of life… 🎶”

File: pull-requests.md
Comment (akash1810) [reply pull-requests.md]: I missed some! https://github.com/guardian/recommendations/pull/110.

**Pull request merged**
